libs/payments/customer/src/lib/invoice.manager.ts

```diff
@@ -260,7 +260,19 @@ export class InvoiceManager {
     let paypalCharge: ChargeResponse;
     try {
       // Charge the PayPal customer after the invoice is finalized to prevent charges with a failed invoice
-      await this.stripeClient.invoicesFinalizeInvoice(invoice.id);
+      try {
+        // Duplicate calls to finalizeInvoice can be made due to race conditions. Failures from re-finalizing an invoice can be ignored.
+        await this.stripeClient.invoicesFinalizeInvoice(invoice.id);
+      } catch (err) {
```

Comment:
AFAIK, it's only with PayPal payments on upgrade where this race condition applies. Instead of wrapping the InvoiceManager method, which could be called in multiple places, could the try catch be moved to the Service which initiates the paypal payment?

Reply:
Ah apologies, this is the right place.